docs/source/upgrade.rst

```diff
@@ -0,0 +1,759 @@
+Note that raw arrays can be directly passed to any :class:`Function`; they are automatically wrapped by :class:`Variable`.
+
+.. _upgrade-snapshot-trigger-removed:
+
+trigger option is removed from snapshot and snapshot_object
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, the ``trigger`` option is removed from the :func:`~training.extensions.snapshot` and :func:`~training.extenisons.snapshot_object` extensions.
+The effect of the option was duplicated with the ``trigger`` option of :meth:`Trainer.extend <training.Trainer.extend>`.
+**If you are passing the** ``trigger`` **argument to these extensions, you have to update your code.**
+The update can be done by passing the value to the corresponding :meth:`Trainer.extend <training.Trainer.extend>`.
+
+.. admonition:: Example
+
+   Assume that ``trainer`` is an instance of :class:`~training.Trainer`, and consider that you were adding a :func:`~training.extensions.snapshot` extension as follows.
+
+   .. code-block:: py
+
+       # Chainer v1
+       trainer.extend(chainer.training.extensions.snapshot(trigger=(1000, 'iteration')))
+
+   It should be updated as follows (note that this code also works with Chainer v1).
+
+   .. code-block:: py
+
+       # Chainer v1/v2
+       trainer.extend(chainer.training.extensions.snapshot(), trigger=(1000, 'iteration'))
+
+.. _upgrade-invoke-before-training-removed:
+
+Extension.invoke_before_training is removed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, The attribute ``invoke_before_training`` of :class:`~training.Extension` is removed.
+Instead, the :meth:`Extension.initialize <training.Extension.initialize>` method is added.
+This method is called by :meth:`Trainer.run <training.Trainer.run>` before entering the training loop.
+
+In Chainer v1, the extension is just called before entering the training loop when ``invoke_before_training`` is ``True``.
+**You need to fix the extension and override** :meth:`~training.Extension.initialize` **if you have custom extensions.**
+If you are using the :func:`~training.make_extension` decorator, you can set the ``initialize`` function by passing ``initializer`` to :func:`~training.make_extension`.
+
+.. _upgrade-dump-graph-only-once:
+
+The dump_graph extension dumps the valid graph only at its first invocation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, the :func:`~training.extensions.dump_graph` extension dumps the valid computational graph only at its first invocation.
+**If you want to dump the graph more than once, you have to fix the code.**
+The easiest fix is setting the ``chainer.config.keep_graph_on_report`` flag to ``True``.
+*Note that this fix will cancel the improvement on the memory consumption made in Chainer v2.*
+More memory-efficient fix is to dump the graph without using an extension, e.g. by customizing the loss function or the updater.
```

Comment:
If you are thinking to merge #2640 before the release of v2, we should add `chainer.config.keep_graph_on_report` to the list of global configurations.